Source/Wikitivity.Agent/WikitivityUploadsAgent.cs

```diff
@@ -51,155 +57,130 @@ public override void Execute()
 			{
 				RaiseMessage("Obtaining Case IDs", 1);
 
-				using (IRSAPIClient proxy = Helper.GetServicesManager().CreateProxy<IRSAPIClient>(ExecutionIdentity.System))
+				Task.Run(async () =>

```

Comment:
this code can be made simpler by just calling wAHelper.GetCasesWithWikitivityOM(proxy).Wait();

Reply:
This a super long function.  It might be worth breaking this up more.
For testing, you should move this into a WikitivityJobExecutor.

You create that jobExecutor in your agent and then call handlejobs on that it.  That way, you can agent base has very little loggic outside of injecting in the requirements to the WikitivityJobExecutor.  Then you can unit test all of this logic because you don't need a work with the agent base, but you work against the WikitivityJobExecutor.